src/transformers/generation/utils.py

```diff
@@ -1778,6 +1778,25 @@ def typeerror():
 
         return result
 
+    def _has_unfinished_sequences(self, this_peer_finished: bool, synced_gpus: bool, device: torch.device) -> bool:
+        """
+        Returns whether there are still unfinished sequences in the device. The existence of unfinished sequences is
+        fed through `this_peer_finished`. ZeRO stage 3-friendly.
+        """
+        if synced_gpus:
+            # Under synced_gpus the `forward` call must continue until all gpus complete their sequence.
+            # The following logic allows an early break if all peers finished generating their sequence
+            this_peer_finished_flag = torch.tensor(0.0 if this_peer_finished else 1.0).to(device)
+            # send 0.0 if we finished, 1.0 otherwise
+            dist.all_reduce(this_peer_finished_flag, op=dist.ReduceOp.SUM)
+            # did all peers finish? the reduced sum will be 0.0 then
+            if this_peer_finished_flag.item() == 0.0:
+                return False
+        else:
+            if this_peer_finished:
+                return False
```

Comment:
nit

```suggestion
        elif this_peer_finished:
            return False
```

Reply:
no good: if `synced_gpus` is `True`, we only want to return `False` when `this_peer_finished is True` in all distributed devices. With this change, we can return `False` when `this_peer_finished is True` in a single device :)